network/wsPeer.go

```diff
@@ -393,6 +429,33 @@ func dedupSafeTag(t protocol.Tag) bool {
 	return t == protocol.AgreementVoteTag || t == protocol.TxnTag
 }
 
+// MaxDecompressedMessageSize defines a maximum decompressed data size
+// to prevent zip bombs
+const MaxDecompressedMessageSize = 20 * 1024 * 1024 // some large enough value
+
+func decompressMsg(data []byte) ([]byte, error) {
+	r := zstd.NewReader(bytes.NewReader(data))
+	b := make([]byte, 0, 1024)
+	for {
+		if len(b) == cap(b) {
+			b = append(b, 0)[:len(b)]
+		}
+		n, err := r.Read(b[len(b):cap(b)])
+		b = b[:len(b)+n]
+		if err != nil {
+			r.Close()
+			if err == io.EOF {
+				return b, nil
+			}
+			return nil, err
+		}
+		if len(b) >= MaxDecompressedMessageSize {
```

Comment:
A bit pedantic, but should this be `len(b) > MaxDecompressedMessageSize`?  Otherwise it's impossible to decompress a message of `MaxDecompressedMessageSize`.

Reply:
Makes sense, fixed